src/unix/haiku/mod.rs

```diff
@@ -0,0 +1,719 @@
+use dox::mem;
+
+cfg_if! {
```

Comment:
I don't really follow the logic there.. If I move all the ifcfg's into something like this:
mod.rs:
```rust
use dox::mem;

cfg_if! {
    if #[cfg(target_pointer_width = "64")] {
        mod b64;
    } else {
        mod b32;
    }
}
```

b32.rs:
```rust
pub type c_long = i32;
pub type c_ulong = u32;

const ULONG_SIZE: usize = 32;
```

Then I begin to get additional style issues:
```
src/unix/haiku/mod.rs:11 - typedef found after module when it belongs before
```